internal/pkg/service/common/etcdop/op/response.go

```diff
@@ -12,16 +14,20 @@ func newRawResponse(client etcd.KV, opts []Option) *RawResponse {
 	return &RawResponse{client: client, options: opts}
 }
 
-func (v RawResponse) Client() etcd.KV {
+func (v *RawResponse) WithOpResponse(r etcd.OpResponse) *RawResponse {
+	clone := *v
+	clone.response = r
+	return &clone
```

Comment:
This clone is just shallow, actually just `response` is copied. Is this correct, we assume that client and options remains the same therefore we done need to copy them.

Reply:
I was just making sure that It is correct. This caused in my past job quite a lot of problems that we copied just partly object and not whole object and we rewrite it through various functions which caused unncessary issues.

Reply:
:+1: I just didn't understand where you see the error.